clients/src/main/java/org/apache/kafka/common/utils/ByteUtils.java

```diff
@@ -150,17 +151,32 @@ public static void writeUnsignedIntLE(byte[] buffer, int offset, int value) {
      * @throws IllegalArgumentException if variable-length value does not terminate after 5 bytes have been read
      */
     public static int readUnsignedVarint(ByteBuffer buffer) {
```

Comment:
Shall we have a comment indicating that we borrowed the implementation from Netty with a link to the relevant file?

Reply:
Sure, will do. 

Also protobuf has a similar unrolled implementation for its c++ code at https://github.com/protocolbuffers/protobuf/blob/2dc5338ea222e1f4e0357e46b702ed6a0e82aaeb/src/google/protobuf/io/coded_stream.cc#L422 (it's Java variant doesn't use the unrolled implementation since they decided to add a different implementation which favours cases when varints are 1 byte).